### PR TITLE
BUGFIX: Parse data 

### PR DESCRIPTION
**Pull request by rmarkins-godaddy**
## Summary

This PR fixes an issue on `object:create` where if the data is already a JSON object we need to parse it prior doing a  `JSON.stringify` to make sure it matches what the dynamoDB tables expect

* I also added a github workflow for cicd and updated the readme to use the proper links

## Changelog

- Parse data prior to post for `object:create`

### CODE REVIEW

File: .github/workflows/cicd.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install Dependencies
+        run: npm ci
+      - name: Run Tests & Lint
+        run: npm test
```
Comment (JacopoDaeli) [inline .github/workflows/cicd.yaml:30]: Add EOL

File: src/commands/object/create.js [hunk 2]
```diff
@@ -38,6 +38,8 @@ class CreateCommand extends BaseCommand {
 
     if (!data) {
       data = await this._readStdin();
+    } else if (typeof data === 'string') {
+      data = JSON.parse(data);
```
Comment (JacopoDaeli) [inline src/commands/object/create.js:42]: First of all, add `try/catch`, in case it's not a JSON.
Also, I doubt this will work.

```
if (!data) {
  data = await this._readStdin();
} 

if (typeof data === 'string') {
  try {
    data = JSON.parse(data);
  } catch (err) {}
}

```

File: README.md [hunk 3]
```diff
@@ -57,7 +57,7 @@ OPTIONS
   --version
 ```
 
-_See code: [src/commands/cdn/index.js](https://github.com/warehouseai/wrhs/blob/v1.1.0/src/commands/cdn/index.js)_
+_See code: [src/commands/cdn/index.js](https://github.com/warehouseai/wrhs/blob/master/src/commands/cdn/index.js)_
```
Comment (JacopoDaeli) [inline README.md:60]: No need to update the readme. It gets auto updated when u generate a new version

File: src/commands/object/create.js (hunk 2 above)
Comment (JacopoDaeli) [reply src/commands/object/create.js:42]: Also, is the request method expecting a JSON object? To be checked

**Review by JacopoDaeli — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (rmarkins-godaddy) [reply README.md:60]: Hmm that's odd because the version you have is `v7.0.0` but this is pointing to `v1.1.0`. Is that done via some github magic? It could be because the package version and the tags are mismatched 😞

**Comment by JacopoDaeli** [comment]
Are u trying to store a JSON object as Map type in Dynamo? Why not storing as string, then parse it backend side when u read the data?

File: src/commands/object/create.js (hunk 2 above)
Comment (rmarkins-godaddy) [reply src/commands/object/create.js:42]: Looking at this line https://github.com/warehouseai/wrhs/blob/master/src/utils/request.js#L58-L76 it is submitting as a JSON, so one would assume it would work w/ a JSON object that is being passed along

**Comment by JacopoDaeli** [comment]
This is not a bug, I would prefer to keep "object data" stored as string in Dynamo. So that we can store arbitrary strings, that could or could not be JSONs.

File: src/commands/object/create.js (hunk 2 above)
Comment (JacopoDaeli) [reply src/commands/object/create.js:42]: It's submitting the entire body. `data` is just a field of that body, and it's string.

File: .github/workflows/cicd.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+
+   
+# Workflow to run CI and tests for all branches on push and on pull requests
```
Comment (msluther) [inline .github/workflows/cicd.yaml:3]: Can you do this as a separate PR please?

File: src/commands/object/create.js (hunk 2 above)
Comment (rmarkins-godaddy) [reply src/commands/object/create.js:42]: sorry my mistake of the line. Though per docs it is expecting a JSON object to be passed as either stdin or `data`.

**Review by JacopoDaeli — APPROVED** [review]
Unblocking

**Review by JacopoDaeli — CHANGES_REQUESTED** [review]
NOT NEEDED

File: src/commands/object/create.js [hunk 5]
```diff
@@ -40,6 +40,14 @@ class CreateCommand extends BaseCommand {
       data = await this._readStdin();
     }
 
+    if (typeof data === 'string') {
+      try {
+        data = JSON.parse(data);
+      } catch (err) {
+        console.error(`Unable to parse due to: ${err}`);
```
Comment (JacopoDaeli) [inline src/commands/object/create.js:47]: This could be expected. Remove. `data` can be just a string.

**Comment by dcousineau-godaddy** [comment]
It is a fairly glaring hole in the DX of this CLI library that the output of one command cannot be used as an input to another command. I've already been struggling very hard to understand why uploading to the via `wrhs cdn:upload <dir> <pkg>` then an `wrhs object:create --version $VERSION --env $ENV --data <output from previous>` for _each_ env.

It seems to me that if `--data` can be an arbitrary string, then `--data` has absolutely no relation to `wrhs cdn:upload ...` especially given that `wrhs cdn:upload ...` dumps out a JSON string of a given structure I assumed was required by `wrhs create:object`. If so I question the utility of this flow entirely.

I have a larger question: does `wrhs cdn:upload ...` create a databases entry, especially given the JSON that's returned?

File: src/commands/object/create.js (hunk 5 above)
Comment (dcousineau-godaddy) [reply src/commands/object/create.js:47]: If data can be an arbitrary string, what is expected then? This seems like an extremely easy case for mistakes and failures to make its way into the system as there would be no validation?

Comment (dcousineau-godaddy) [reply src/commands/object/create.js]: More importantly, what happens when an arbitary string is uploaded? What use case is expected here?

**Review by JacopoDaeli — APPROVED** [review]
LGTM
